test/jdk/java/net/httpclient/EmptyAuthenticate.java

```diff
@@ -24,55 +24,118 @@
-        server.start();
+import static java.net.http.HttpClient.Builder.NO_PROXY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
-        HttpResponse<String> response = null;
-        //run client:
+class EmptyAuthenticate {
+
+    private static final SSLContext SSL_CONTEXT = createSslContext();
+
+    private static final String WWW_AUTH_HEADER_NAME = "WWW-Authenticate";
+
+    private static SSLContext createSslContext() {
         try {
-            HttpClient httpClient = HttpClient.newHttpClient();
-            HttpRequest request = HttpRequest.newBuilder(new URI("http://localhost:" + port + "/")).GET().build();
-            //this line will throw NPE (wrapped by IOException) when parsing empty www-authenticate response header in AuthenticationFilter:
-            response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
-            boolean ok = !response.headers().firstValue("WWW-Authenticate").isEmpty();
-            if (!ok) {
-                throw new RuntimeException("WWW-Authenicate missing");
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
-            throw new RuntimeException("Test failed");
+            return new SimpleSSLContext().get();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("args")
+    void test(Version version, boolean secure) throws Exception {
+        HttpTestServer server = createStartedServerRespondingWithEmptyWwwAuthHeader(version, secure);
+        try (HttpClient client = createClient(version, secure)) {
+            HttpRequest request = createRequest(server, secure);
+            HttpResponse<Void> response = client.send(request, HttpResponse.BodyHandlers.discarding());
+            HttpHeaders responseHeaders = response.headers();
+            assertEquals(
+                    "",
+                    responseHeaders.firstValue(WWW_AUTH_HEADER_NAME).orElse(null),
+                    () -> "was expecting empty `%s` header in: %s".formatted(
+                            WWW_AUTH_HEADER_NAME, responseHeaders.map()));
         } finally {
-            server.stop(0);
+            server.stop();
         }
     }
+
+    static Stream<Arguments> args() {
+        return Stream
+                .of(Version.HTTP_1_1, Version.HTTP_2)
+                .flatMap(version -> Stream
+                        .of(true, false)
+                        .map(secure -> Arguments.of(version, secure)));
+    }
+
+    private static HttpTestServer createStartedServerRespondingWithEmptyWwwAuthHeader(Version version, boolean secure)
+            throws IOException {
+        HttpTestServer server = secure
+                ? HttpTestServer.create(version, SSL_CONTEXT)
+                : HttpTestServer.create(version);
+        HttpTestHandler handler = new ServerHandlerRespondingWithEmptyWwwAuthHeader();
+        server.addHandler(handler, "/");
```

Comment:
```suggestion
        server.addHandler(handler, "/EmptyAuthenticate/");
```

Reply:
Applied it in 7b5c849002c2302a3b4d4cdec3fd427fc9a1c4f7.